synapseclient/credentials/cached_sessions.py

```diff
@@ -0,0 +1,79 @@
+import keyring
+import os
+import json
+import warnings
+from keyring.errors import PasswordDeleteError
+from keyring.backends.fail import Keyring as FailKeyring
+
+
+SYNAPSE_CACHED_SESSION_APLICATION_NAME = "SYNAPSE.ORG_CLIENT"
+SESSION_CAHCE_FILEPATH = os.path.expanduser("~/.synapseSession")
+
+# In the case where no key store backend is available (most likely in Linux), a fail.Keyring is returned which will throw errors when any of its functions are called
+# However, the errors thrown are of the generic type RuntimeError (not a subclass of RuntimeError).
+# It didn't feel safe to try/except and ignore RuntimeError since there are very many other RuntimeErrors that could occur.
+# Instead, we use this boolean to keep track of whether we should call the keyring methods
+_keyring_is_available = not isinstance(keyring.get_keyring(), FailKeyring)
```

Comment:
reword this section to explain what `_keyring_is_available` is being used for instead of how we decided to use it.